Products/Zuul/tests/test_devicefacade.py

```diff
@@ -167,15 +166,82 @@ def test_deviceSortByProdState(self):
         results = self.facade.getDeviceBrains(uid="/zport/dmd/Devices", sort='productionState')
         resultIter = iter(results)
         self.assertEquals(2, results.total)
-        self.assertEquals(resultIter.next().getObject().getProductionState(), 400)
-        self.assertEquals(resultIter.next().getObject().getProductionState(), 1000)
+        self.assertEquals(resultIter.next().getProductionState(), 400)
+        self.assertEquals(resultIter.next().getProductionState(), 1000)
 
         # descending order
         results = self.facade.getDeviceBrains(uid="/zport/dmd/Devices", sort='productionState', dir='DESC')
         resultIter = iter(results)
         self.assertEquals(2, results.total)
-        self.assertEquals(resultIter.next().getObject().getProductionState(), 1000)
-        self.assertEquals(resultIter.next().getObject().getProductionState(), 400)
+        self.assertEquals(resultIter.next().getProductionState(), 1000)
+        self.assertEquals(resultIter.next().getProductionState(), 400)
+
+    def test_deviceSearchByProdStateAndLocationReturnsCorrectDevices(self):
+        manage_addLocation(self.dmd.Locations, "test1")
+
+        devMaintenance = self.dmd.Devices.createInstance('devMaintenance')
+        devMaintenance.setLocation("/test1")
+        devMaintenance.setProdState(400)
+
+        devProduction = self.dmd.Devices.createInstance('devProduction')
+        devProduction.setLocation("/test1")
+        devProduction.setProdState(1000)
+
+        # this device should never be returned
+        devOther = self.dmd.Devices.createInstance('devOther')
+        devOther.setLocation("/test1")
+        devOther.setProdState(1)
+
+        results = self.facade.getDeviceBrains(uid="/zport/dmd/Devices", params=dict(productionState=[400], location="test1"))
+        self.assertEquals(1, results.total)
+        device = iter(results).next()
+        self.assertEquals(device.getProductionState(), 400)
+        self.assertEquals(device.location.getRelatedId(), "test1")
+
+        results = self.facade.getDeviceBrains(uid="/zport/dmd/Devices", params=dict(productionState=[1000], location="test1"))
+        self.assertEquals(1, results.total)
+        device = iter(results).next()
+        self.assertEquals(device.getProductionState(), 1000)
+        self.assertEquals(device.location.getRelatedId(), "test1")
+
+    def test_deviceSortByProdStateWithLocationFilterReturnsCorrectDevicesAndSortsCorrectly(self):
```

Comment:
We need a test where the sort is by an unindexed field and the filter is productionState, as described in ZEN-26901